src/System.Net.Security/src/System/Net/_SslSessionsCache.cs

```diff
@@ -96,6 +100,11 @@ public override bool Equals(Object y)
                     return false;
                 }
 
+                if (isServerMode != other.isServerMode)
```

Comment:
Now I understand how this works but I don't understand the reason for this getting changed from the original implementation.
The object's now bigger and requires extra operations to determine equality.